### PR TITLE
chore(i18n): Update and correct Japanese translation

### PR DESCRIPTION
**Pull request by ma10**
Update the Japanese translation based on the message catalog as of 2023-05-11.
While adding translation for newly added rule, I also reviewed the whole file, and correct the existing translations.

* some of the messages were misleading
* made the wording consistent

### CODE REVIEW

**Comment by ma10** [comment]
@kiyochika Could you review this one?

File: locales/ja.json [hunk 1]
```diff
@@ -45,6 +49,10 @@
       "description": "すべてのARIA progressbarノードにアクセシブルな名前があることを確認します",
       "help": "ARIA progressbarノードにはアクセシブルな名前がなければなりません"
     },
+    "aria-prohibited-attr": {
+      "description": "Ensures ARIA attributes are not prohibited for an element's role",
+      "help": "Elements must only use permitted ARIA attributes"
+    },
```
Comment (WilcoFiers) [inline locales/ja.json:55]: @ma10 Did you mean to leave these without a translation?

Comment (ma10) [reply locales/ja.json:55]: No, it should be something that has been added to the base branch after my last update and I overlooked  when I merged the base branch last time.
I'll try to work on that one as soon as I get a chance.

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: locales/ja.json [hunk 2]
```diff
@@ -59,26 +67,26 @@
...
+      "help": "aria-roledescriptionはセマンティックなロールを持った要素に使用しなければなりません"
     },
     "aria-roles": {
-      "description": "すべてのロール属性が指定された要素で、有効な値が使用されていることを確認します",
+      "description": "すべてのrole属性が指定された要素で、有効な値が使用されていることを確認します",
       "help": "使用されているARIAロールは有効な値に一致しなければなりません"
     },
     "aria-text": {
-      "description": "role=\"text\"が指定されている要素にフォーカス可能な子孫がないことを確認します",
+      "description": "\"role=text\"が指定されている要素にフォーカス可能な子孫がないことを確認します",
```
[the remark this answers is not in the document]
Comment (ma10) [reply locales/ja.json:77]: Yes, I thought the quotes should be around the value, but the file lib/rules/aria-text.json has it this way.
I just created a pull request #4131 to fix this.

File: locales/ja.json [hunk 3]
```diff
@@ -412,80 +420,88 @@
...
-        "singular": "ARIA ロール ${data.values} は指定された要素に許可されていません",
-        "plural": "ARIA ロール ${data.values} は指定された要素に許可されていません"
+        "singular": "ARIAロール${data.values}は指定された要素では許可されていません",
+        "plural": "ARIAロール${data.values}は指定された要素では許可されていません"
       },
       "incomplete": {
-        "singular": "ARIA ロール ${data.values} この要素に許可されていないため、要素が表示されたときはARIAロールを削除する必要があります",
-        "plural": "ARIA ロール ${data.values} この要素に許可されていないため、要素が表示されたときはARIAロールを削除する必要があります"
+        "singular": "ARIAロール${data.values}はこの要素で許可されていないため、要素が表示されたときはARIAロールを削除する必要があります",
+        "plural": "ARIAロール${data.values}この要素に許可されていないため、要素が表示されたときはARIAロールを削除する必要があります"
```
[the remark this answers is not in the document]
Comment (ma10) [reply locales/ja.json:433]: In Japanese, many nouns don't have different forms for singular and plural (there are suffixes to indicate plural for certain nouns, but not for all nouns.)
This case is no exception and the singular and the plural should be the same.
While reviewing your suggestion, I found a typo, so I'll commit a fix.

File: locales/ja.json (hunk 1 above)
Comment (ma10) [reply locales/ja.json]: Done with commit 1c2d6695

File: locales/ja.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (ma10) [reply locales/ja.json]: Done with the commit 141ba17fbe

File: locales/ja.json [hunk 4]
```diff
@@ -537,6 +545,18 @@
         "plural": "無効なARIA属性名です: ${data.values}"
       }
     },
+    "braille-label-equivalent": {
+      "pass": "aria-braillelabelはアクセシブルなテキストがある要素に対して使用されています",
+      "fail": "aria-braillelabelがアクセシブルなテキストがない要素に対して使用されています",
+      "incomplete": "アクセシブルなテキストを算出することができません"
+    },
+    "braille-roledescription-equivalent": {
+      "pass": "aria-brailleroledescriptionはアクセシブルなテキストがある要素に対して使用されています",
```
Comment (WilcoFiers) [inline locales/ja.json:554]: Looks like another issue with the original. PR with the fix here: https://github.com/dequelabs/axe-core/pull/4133 Could you update this to match @ma10?

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: locales/ja.json (hunk 4 above)
Comment (ma10) [reply locales/ja.json]: Done in commit d80a727b

**Review by WilcoFiers — APPROVED** [review]

**Comment by WilcoFiers** [comment]
Reviewed for security.

**Comment by ma10** [comment]
@WilcoFiers I appreciate your time and effort for the review.

**Pull request merged**